src/extensions/extension-loader/extension-loader.ts

```diff
@@ -339,21 +341,32 @@ export class ExtensionLoader {
       // Remove null values
       .filter(extension => Boolean(extension));
 
-    // We first need to wait until each extension's `onActivate` is resolved,
+    // We first need to wait until each extension's `onActivate` is resolved or rejected,
     // as this might register new catalog categories. Afterwards we can safely .enable the extension.
-    await Promise.all(extensions.map(extension => extension.activated));
+    await Promise.all(
+      extensions.map(extension =>
+        // If extension activation fails, log error
+        extension.activated.catch((error) => {
+          extension.activationError = error;
+          logger.error(`${logModule}: activation extension error`, { ext: extension.installedExtension, error });
+          this.instances.delete(extension.extId);
```

Comment:
Also, why delete the instances?

Reply:
You're right, we can keep extensions that can't be enabled in `.instances`. That mirrors the original functionality too. Fixed